sycl/include/sycl/builtins.hpp

```diff
@@ -2164,6 +2206,47 @@ select(T a, T b, T2 c) __NOEXC {
   return __sycl_std::__invoke_select<T>(a, b, c);
 }
 
+// other marray relational functions
+
+template <typename T, size_t N>
+detail::enable_if_t<detail::is_sigeninteger<T>::value, detail::anyall_ret_t>
+any(marray<T, N> x) __NOEXC {
+  for (int i = 0; i < N; i++)
+    if (any(x[i]))
+      return true;
+  return false;
```

Comment:
`marray` has `begin`/`end`, so we can use `std::any` (preferably) or at least range-based for loop.

Reply:
Applied in https://github.com/intel/llvm/pull/9110/commits/dd017157c35bf89331f71cf1799b1e083a46330d